frame/elections-phragmen/src/lib.rs

```diff
@@ -334,26 +407,56 @@ decl_module! {
 		///
 		/// A defunct voter is defined to be:
 		///   - a voter whose current submitted votes are all invalid. i.e. all of them are no
-		///     longer a candidate nor an active member.
+		///     longer a candidate nor an active member or a runner-up.
+		///
+		///
+		/// The origin must provide the number of current candidates for the purpose of accurate
+		/// weight calculation.
 		///
 		/// # <weight>
-		/// #### State
-		/// Reads: O(NLogM) given M current candidates and N votes for `target`.
-		/// Writes: O(1)
+		/// No Base weight based on min square analysis.
+		/// State reads:
+		///  	- Voting(reporter)
+		///  	- Candidate.len()
+		///  	- Voting(Target)
+		///  	- Candidates, Members, RunnersUp (is_defunct_voter) [and Voting(Target) -- already counted]
+		/// State writes:
+		/// 	- Lock(reporter || target)
+		/// 	- AccountBalance(reporter) + AccountBalance(target)
+		/// 	- Voting(reporter || target)
+		/// Note: the db access is worse with respect to db, which is when the report is correct.
```

Comment:
There is seriously no way to know if the report is correct or not without executing the whole damn thing, hence we assume worse case here. The diff is merely one db read extra.